### PR TITLE
Nc news 6

### PR DESCRIPTION
**Pull request by kctan0814**
Finished writing test and code to make sure endpoint is working as intended. Accounting for edge cases and sad path.

### CODE REVIEW

File: app/__tests__/app.test.js [hunk 1]
```diff
@@ -24,4 +24,43 @@ describe('GET /api/topics', () => {
...
+        expect(comments.length).toBe(11)
+        expect(Array.isArray(comments)).toBe(true)
+        comments.forEach(comment => {
+          expect(comment).toMatchObject({
+            comment_id: expect.any(Number),
+            votes: expect.any(Number),
+            created_at: expect.any(String),
+            author: expect.any(String),
+            body: expect.any(String),
+            article_id: expect.any(Number)
```
Comment (lewispricey) [inline app/__tests__/app.test.js:44]: Since you're looking for a specfic articles comments you could be more explicit with what article_id you'd expect here 😄

File: app/__tests__/app.test.js [hunk 2]
```diff
@@ -24,4 +24,43 @@ describe('GET /api/topics', () => {
...
+  test('200: returns an empty array if article exist with no comments', () => {
+    return request(app)
+      .get('/api/articles/4/comments')
+      .expect(200)
+      .then(({body: {comments}}) => {
+        expect(comments.length).toBe(0)
+        expect(Array.isArray(comments)).toBe(true)
+      })
+  })
+  test('400: retruns an error message of "Not foumd" when passed with an article id not in the database', () => {
```
Comment (lewispricey) [inline app/__tests__/app.test.js:58]: This test expectation is correct we'd expect a 404 here - just a description update needed.

File: app/__tests__/app.test.js [hunk 3]
```diff
@@ -24,4 +24,43 @@ describe('GET /api/topics', () => {
...
+            author: expect.any(String),
+            body: expect.any(String),
+            article_id: expect.any(Number)
+          })
+        })
+      })
+  })
+  test('200: returns an empty array if article exist with no comments', () => {
+    return request(app)
+      .get('/api/articles/4/comments')
```
Comment (lewispricey) [inline app/__tests__/app.test.js:51]: I really like that you've included this test since an id that exists but has no comments is definitely empty array behaviour.

File: app/model/articles.model.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
+const db = require('../../db/connection')
+
+exports.selectCommentsByArticleId = (id) => {
+  return db.query("SELECT * FROM articles WHERE article_id = $1", [id])
```
Comment (lewispricey) [inline app/model/articles.model.js:4]: You might already have a model written that performs this query and rejects an article that doesnt exit so you don't have to have another copy of this query 👀

**Review by lewispricey — CHANGES_REQUESTED** [review]
Great work!

I'd say theres just one test case you're missing here which would be a 400 - for an invalid id like /banana

Please could you add that in and pop over another PR :)
